compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -3436,11 +3436,28 @@ public BLangNode transform(ArrayTypeDescriptorNode arrayTypeDescriptorNode) {
             } else {
                 Node keyExpr = arrayTypeDescriptorNode.arrayLength().get();
                 if (keyExpr.kind() == SyntaxKind.NUMERIC_LITERAL) {
+                    int length;
                     Token literalToken = ((BasicLiteralNode) keyExpr).literalToken();
                     if (literalToken.kind() == SyntaxKind.DECIMAL_INTEGER_LITERAL_TOKEN) {
-                        sizes.add(new BLangLiteral(Integer.parseInt(literalToken.text()), symTable.intType));
+                        long lengthCheck = Long.parseLong(literalToken.text());
```

Comment:
What if the long is so long that this method throws `NumberFormatException`
https://docs.oracle.com/javase/7/docs/api/java/lang/Long.html#parseLong(java.lang.String)

Reply:
I'll change it into double instead of long.